lib/WTSI/NPG/Data/BamDeletion.pm

```diff
@@ -0,0 +1,325 @@
+package WTSI::NPG::Data::BamDeletion;
+
+use Moose;
+#use English qw{-no_match_vars};
+use POSIX qw[strftime];
+use Cwd;
+use File::Slurp;
+use File::Temp qw[tempdir];
+use File::Basename;
+use Readonly;
+use Carp;
+
+
+use WTSI::NPG::HTS::Illumina::AlnDataObject;
+use WTSI::NPG::iRODS::Publisher;
+
+with qw{
+  MooseX::Getopt
+  WTSI::DNAP::Utilities::Loggable
+       };
+
+our $VERSION = '0';
+
+our $DATE = strftime '%Y%m%d', localtime;
+our $DELETION_MSG = qq[#The data from this file was removed on $DATE as consent was withdrawn];
+
+=head1 NAME
+
+WTSI::NPG::Data::BamDeletion
+
+=head1 SYNOPSIS
+
+=head1 DESCRIPTION
+
+Update specified iRODS files leaving behind a header file for alignment files and a stub mentioning consent withdrawn for other files.  File name remains the same as the original.
+
+The original meta data is attached and the RT ticket added.
+
+For use with e.g. iRODS files identified with npg_process_consent_withdrawn.pl
+
+=head1 SUBROUTINES/METHODS
+
+=head2 dry_run
+
+Dry run flag, false by default. No changes to iRODS data
+
+=cut
+
+has 'dry_run' => (
+  isa           => 'Bool',
+  is            => 'ro',
+  required      => 0,
+  default       => 1,
+  documentation => 'dry run flag',
+);
+
+
+=head2 irods
+
+WTSI::NPG::iRODS type object mediating access to iRODS,
+required.
+
+=cut
+
+has 'irods' => (
+  isa        => 'WTSI::NPG::iRODS',
+  is         => 'ro',
+  required   => 1,
+  metaclass => 'NoGetopt',
+);
+
+=head2 rt_ticket
+
+=cut
+
+has 'rt_ticket' => (
+  isa        => 'Int',
+  is         => 'ro',
+  required   => 0,
+  documentation=>'RT ticket number as integer, an optional argument'
+);
+
+=head2 input_fofn
+
+File of File Names  specified
+
+=cut
+
+has 'input_fofn' => (
+  isa        => 'Str',
+  is         => 'ro',
+  required   => 0,
+  documentation=>'File of iRODS file name paths'
+);
+
+=head2 file
+
+File to be removed 
+
+=cut
+
+has 'file' => (
+  isa        => 'Str',
+  is         => 'ro',
+  required   => 1,
+  documentation=>'File to be removed from iRODS and replaced with stub file with meta data retained'
+);
+
+=head2 md5_file
+
+=cut
+
+has 'md5_file' => (
+  isa        => 'Str',
+  is         => 'ro',
+  required   => 0,
+  lazy_build =>1
+);
+sub _build_md5_file {
+    my $self = shift;
+    my $filename = fileparse($self->file);
+    my $md5file  = $self->outdir.qq[/$filename].q[.md5];
+return $md5file;
+}
+
+sub _build_md5sum {
```

Comment:
This method has its own test, but no documentation on the method or on the test.

Reply:
md5sum attribute added with this as a builder  & documentation